include/mata/nfa.hh

```diff
@@ -1310,13 +1326,98 @@ public:
     std::list<Symbol> get_complement(const std::set<Symbol>& syms) const override;
 
     /**
-     * Add new symbol to the alphabet.
-     * @param key User-space representation of the symbol.
-     * @param value Number of the symbol to be used on transitions.
+     * @brief Add new symbol to the alphabet.
+     *
+     * Throws an exception when the adding fails.
+     *
+     * @param[in] key User-space representation of the symbol.
+     * @param[in] value Number of the symbol to be used on transitions.
+     * @return Result of the insertion as @c InsertionResult.
+     */
+     InsertionResult add_new_symbol(const std::string& key, Symbol value) {
+        const InsertionResult insertion_result{ try_add_new_symbol(key, value) };
+        if (!insertion_result.second) { // If the insertion of key-value pair failed.
+            throw std::runtime_error("multiple occurrence of the same symbol");
+        }
+        update_next_symbol_value(value);
+        return insertion_result;
+    }
+
+    /**
+     * @brief Try to add symbol to the alphabet map.
+     *
+     * Does not throw an exception when the adding fails.
+     *
+     * @param[in] key User-space representation of the symbol.
+     * @param[in] value Number of the symbol to be used on transitions.
+     * @return Result of the insertion as @c InsertionResult.
      */
-    void add_symbol(const std::string& key, Symbol value) { symbol_map.insert({key, value}); }
+    InsertionResult try_add_new_symbol(const std::string& key, Symbol value) {
```

Comment:
I suggest `try_to_add_new_symbol` or `try_adding_new_symbol`. It's not much longer, and it's grammatically correct :>

Reply:
I agree. I will change the name to `try_to_add_new_symbol()`.